lib/index.ts

```diff
@@ -31,64 +32,82 @@ export * from './local-image';
 export function splitBuildStream(
 	composition: Compose.Composition,
 	buildStream: Stream.Readable,
-): Promise<BuildTask[]> {
+): Bluebird<BuildTask[]> {
 	const images = Compose.parse(composition);
 	return fromImageDescriptors(images, buildStream);
 }
 
-export function fromImageDescriptors(
-	images: Compose.ImageDescriptor[],
-	buildStream: Stream.Readable,
-): Promise<BuildTask[]> {
+function setDefault<T>(obj: { [ k: string ]: T }, key: string, defaultValue: T): T {
+	// Like Python's `dict.setdefault`
```

Comment:
I'd rather an explanation about what it actually does :P

Reply:
Sounds fair , updated :)